geonode/security/utils.py

```diff
@@ -523,6 +523,9 @@ def safe_remove(perms, perm): perms.remove(perm) if perm in perms else None
                     if not AdvancedSecurityWorkflowManager.is_simple_publishing_workflow() and (_resource.is_approved or _resource.is_published):
                         safe_remove(prev_perms, 'change_resourcebase')
                         safe_remove(prev_perms, 'change_resourcebase_metadata')
+                        if _resource.polymorphic_ctype.model == "dataset":
```

Comment:
@mattiagiupponi just as a side note, in master now we have also the `resource_type` and `subtype` fields which already collect those info without the need of using the `polymorphic_ctype.model`.
Generally speaking, mostly for the readability of the code and to try being coherent everywhere, it would be better to use that ones, but it is ok to use the `polymorphic_ctype.model` too.